tests/analyser/analyserunits.cpp

```diff
@@ -0,0 +1,727 @@
+            "The units in '" + op + "(cCst, 5.0)' in equation 'c = " + op + "(cCst, 5.0)' in component 'main' are not equivalent. 'cCst' is in 'volt' while '5.0' is 'dimensionless'.",
+            "The units in '" + op + "(dCst, 7.0)' in equation 'd = " + op + "(dCst, 7.0)' in component 'main' are not equivalent. 'dCst' is in 'frog' while '7.0' is 'dimensionless'.",
+            "The units in '" + op + "(eCst, 9.0)' in equation 'e = " + op + "(eCst, 9.0)' in component 'main' are not equivalent. 'eCst' is in 'imaginary' while '9.0' is 'dimensionless'.",
+            "The units in '" + op + "(fCst, 11.0)' in equation 'f = " + op + "(fCst, 11.0)' in component 'main' are not equivalent. 'fCst' is in 'second' while '11.0' is in 'imaginary'.",
+            "The units in '" + op + "(13.0, 15.0)' in '" + op + "(gCst, " + op + "(13.0, 15.0))' in equation 'g = " + op + "(gCst, " + op + "(13.0, 15.0))' in component 'main' are not equivalent. '13.0' is in 'frog' while '15.0' is in 'imaginary'.",
+        };
+
+        auto analyser = libcellml::Analyser::create();
+
+        analyser->analyseModel(model);
+
+        EXPECT_EQ_ISSUES(expectedIssues, analyser);
+    }
+}
+
+TEST(AnalyserUnits, rem)
+{
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("analyser/units/rem.cellml"));
+
+    EXPECT_EQ(size_t(0), parser->issueCount());
+
+    const std::vector<std::string> expectedIssues = {
+        "The units in 'rem(bCst, 3.0)' in equation 'b = rem(bCst, 3.0)' in component 'main' are not equivalent. 'bCst' is in 'second' while '3.0' is 'dimensionless'.",
+        "The units in 'rem(cCst, 5.0)' in equation 'c = rem(cCst, 5.0)' in component 'main' are not equivalent. 'cCst' is in 'volt' while '5.0' is 'dimensionless'.",
+        "The units in 'rem(dCst, 7.0)' in equation 'd = rem(dCst, 7.0)' in component 'main' are not equivalent. 'dCst' is in 'frog' while '7.0' is 'dimensionless'.",
+        "The units in 'rem(eCst, 9.0)' in equation 'e = rem(eCst, 9.0)' in component 'main' are not equivalent. 'eCst' is in 'imaginary' while '9.0' is 'dimensionless'.",
+        "The units in 'rem(fCst, 11.0)' in equation 'f = rem(fCst, 11.0)' in component 'main' are not equivalent. 'fCst' is in 'second' while '11.0' is in 'imaginary'.",
+    };
+
+    auto analyser = libcellml::Analyser::create();
+
+    analyser->analyseModel(model);
+
+    EXPECT_EQ_ISSUES(expectedIssues, analyser);
+}
+
+TEST(AnalyserUnits, trigonometricOperators)
+{
+    std::vector<std::string> operators = {
+        // clang-format off
+        "sin", "cos", "tan", "sec", "csc", "cot",
+        "sinh", "cosh", "tanh", "sech", "csch", "coth",
+        "arcsin", "arccos", "arctan", "arcsec", "arccsc", "arccot",
+        "arcsinh", "arccosh", "arctanh", "arcsech", "arccsch", "arccoth",
+        // clang-format on
```

Comment:
I can't agree with the use of the clang-format directive here (and in other places in this file).  This is purely for personal taste, which is exactly what using clang-format is working to avoid.

Reply:
Done in commit 2bec85f0.